src/nlp/arrow_dataset.py

```diff
@@ -118,12 +125,14 @@ def __init__(
         info: Optional[Any] = None,
         split: Optional[Any] = None,
     ):
-        super().__init__(info=info, split=split)
+        DatasetInfoMixin.__init__(self, info=info, split=split)
+        IndexableMixin.__init__(self)
```

Comment:
I'm a tiny bit worried about this multiple inheritance scenario but ok, let's see how it goes for now.